### PR TITLE
[Polaris Lighthouse] Deduped stylelint-polaris category rules

### PR DESCRIPTION
**Pull request by sophschneider**
### WHY are these changes introduced?

There were rules in multiple categoires

### WHAT is this pull request doing?

Deleting redundant rules that were in the wrong category

### CODE REVIEW

File: stylelint-polaris/index.js [hunk 1]
```diff
@@ -78,7 +78,6 @@ const stylelintPolarisCoverageOptions = {
         /\$skeleton-shimmer-duration/,
         /\$easing-data/,
         // Legacy custom properties
-        /--p-range-slider-thumb-scale/,
```
Comment (sophschneider) [inline stylelint-polaris/index.js:81]: moved this with the other range slider in layout, not sure how it got into motion here

File: stylelint-polaris/index.js [hunk 2]
```diff
@@ -187,12 +186,9 @@ const stylelintPolarisCoverageOptions = {
         'thumbnail-size',
         'icon-size',
         'top-bar-height',
-        'z-index',
```
Comment (sophschneider) [inline stylelint-polaris/index.js:190]: This is already in the z-index category, they shouldn't be duplicated here in layout

File: stylelint-polaris/index.js [hunk 3]
```diff
@@ -187,12 +186,9 @@ const stylelintPolarisCoverageOptions = {
...
       ].map(matchNameRegExp),
       'polaris/at-rule-disallowed-list': {
         include: [
-          'hidden-when-printing',
-          'print-hidden',
```
Comment (sophschneider) [inline stylelint-polaris/index.js:195]: These are already in the media category, they shouldn't be duplicated here in layout

File: stylelint-polaris/index.js [hunk 4]
```diff
@@ -210,8 +206,8 @@ const stylelintPolarisCoverageOptions = {
         // Legacy custom properties
         /--p-range-slider-thumb-size-base/,
         /--p-range-slider-thumb-size-active/,
+        /--p-range-slider-thumb-scale/,
         /--p-override-visible/,
-        /--p-override-loading-z-index/,
```
Comment (sophschneider) [inline stylelint-polaris/index.js:214]: This is already in the z-index category, they shouldn't be duplicated here in layout

File: stylelint-polaris/index.js [hunk 5]
```diff
@@ -312,8 +308,6 @@ const stylelintPolarisCoverageOptions = {
       'polaris/global-disallowed-list': [
         // Legacy mixin map-get data
         /\$shadows-data/,
-        /\$fixed-element-stacking-order/,
-        /\$global-elements/,
```
Comment (sophschneider) [inline stylelint-polaris/index.js:316]: These are already in the z-index category, they shouldn't be duplicated here in layout

**Review by chloerice — APPROVED** [review]

**Pull request merged**